### PR TITLE
Fix Meta.enso_project

### LINKED ISSUE #9845: `Enso_Project.enso_project` does not work as advertised
The documentation says that it

```
Returns the Enso project description for the project that the engine was
executed with, i.e., the project that contains the `main` method
```

however in practice, which can be easily confirmed by checking the code - it returns the project associated with the immediate call site.

That means that if I call `Enso_Project.enso_project` from within the `Standard.Base` library, it will actually return the project for `Base`, not the project that is currently running and only calling into `Base`.

Moreover, the result is cached, so the behaviour of this method depends on where it was called first, for example:

1. Create the following file in `Base/0.0.0-dev/src/My_Test.enso`:
   ```
   import project.Meta.Enso_Project

   foo = Enso_Project.enso_project
   ```
2. Rebuild the engine and run a project with following `Main.enso` method:
   ```
   from Standard.Base import all
   import Standard.Base.Meta.Enso_Project
   import Standard.Base.My_Test

   main = 
       IO.println My_Test.foo.name
       IO.println Enso_Project.enso_project.name
   ```
3. Then run it again swapping the last 2 lines:
   ```
       IO.println Enso_Project.enso_project.name
       IO.println My_Test.foo.name
   ```

The first run yields:
```
Base
Base
```
and the second run yields:
```
testproject1
testproject1
```

The result depends only on the place where `Enso_Project.enso_project` was called first.

### PR DESCRIPTION
**Pull request by Akirathan**
Fixes #9845

### Pull Request Description

Fixes `Standard.Base.Meta.Enso_Project.enso_project` to return a project descriptor for the *main* project, i.e., the one configured as a *root* for the engine.

### Important Notes

`enso_project` builtin no longer iterates the stack frames to infer the project descriptor. It derives it from the default package repository.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.

### CODE REVIEW

File: test/Base_Tests/src/Semantic/Meta_Location_Spec.enso [hunk 1]
```diff
@@ -14,15 +16,15 @@ add_specs suite_builder = suite_builder.group "Meta-Value Inspection" group_buil
...
+        loc = "Meta_Location_Spec.enso:18:15-40"
         src.take (Last loc.length) . should_equal loc
 
     group_builder.specify "should allow to get qualified type names of values" <|
         x = 42
         y = My_Type.Value 1 2 3
         Meta.get_qualified_type_name x . should_equal "Standard.Base.Data.Numbers.Integer"
         Meta.get_simple_type_name x . should_equal "Integer"
-        Meta.get_qualified_type_name y . should_equal "enso_dev.Base_Tests.Semantic.Meta_Location_Spec.My_Type"
+        Meta.get_qualified_type_name y . should_end_with "Meta_Location_Spec.My_Type"
```
Comment (Akirathan) [inline test/Base_Tests/src/Semantic/Meta_Location_Spec.enso:27]: Unrelated change. It ensures that one can run this test as standalone with `enso --run test/Base_Tests/src/Semantic/Meta_Location_Spec.enso`

**Review by 4e6 — APPROVED** [review]

File: test/Base_Tests/src/Semantic/Meta_Location_Spec.enso (hunk 1 above)
Comment (radeusgd) [reply test/Base_Tests/src/Semantic/Meta_Location_Spec.enso:27]: Won't the tests below break if run without `--in-project` anyway?

**Review by radeusgd — APPROVED** [review]
Looks great, I especially appreciate the `ensoProjectWorksInTwoProjects` test.

Thanks for fixing this and unblocking #9875

File: test/Base_Tests/src/Semantic/Meta_Location_Spec.enso (hunk 1 above)
Comment (Akirathan) [reply test/Base_Tests/src/Semantic/Meta_Location_Spec.enso:27]: There is no need to specify `--in-project` anymore since https://github.com/enso-org/enso/pull/8775

Comment (radeusgd) [reply test/Base_Tests/src/Semantic/Meta_Location_Spec.enso:27]: So why was the standalone run ever failing? It should always return `enso_dev.Base_Tests.Semantic.Meta_Location_Spec.My_Type` as the qualified name, no?

If not, what does it return instead??

**Review by JaroslavTulach — APPROVED** [review]

File: test/Base_Tests/src/Semantic/Meta_Location_Spec.enso (hunk 1 above)
Comment (Akirathan) [reply test/Base_Tests/src/Semantic/Meta_Location_Spec.enso:27]: > If not, what does it return instead??

It incorrectly returns `Meta_Location_Spec.My_Type`.

`Meta.get_qualified_type_name` seems broken. Tracked in https://github.com/enso-org/enso/issues/10228. In that issue, there is a task to revert this change once that is fixed.

Comment (radeusgd) [reply test/Base_Tests/src/Semantic/Meta_Location_Spec.enso:27]: Ok, great
